File: Python/Petshop.py
class Petshop:
    # private attribute for Petshop
    __ID = ""
    __nama = ""
    __kategori = ""
    __harga = 0

    def __init__(self, input_id="", input_nama="", input_kategori="", input_harga=0):
        self.__ID = input_id  # Private attribute
        self.__nama = input_nama  # Private attribute
        self.__kategori = input_kategori  # Private attribute
        self.__harga = input_harga  # Private attribute

    # Metode untuk mengambil/mengatur ID
    def get_ID(self):
        return self.__ID

    def set_ID(self, input_id):
        self.__ID = input_id

    # Metode untuk mengambil/mengatur nama
    def get_nama(self):
        return self.__nama

    def set_nama(self, input_nama):
        self.__nama = input_nama

    # Metode untuk mengambil/mengatur kategori
    def get_kategori(self):
        return self.__kategori

    def set_kategori(self, input_kategori):
        self.__kategori = input_kategori

    # Metode untuk mengambil/mengatur harga
    def get_harga(self):
        return self.__harga

    def set_harga(self, input_harga):
        if (
            isinstance(input_harga, int) and input_harga >= 0
        ):  # input_harga diperiksa apakah integer dan lebih dari sama dengan 0
            self.__harga = input_harga
        else:  # jika tidak maka output error
            raise ValueError("Harga must be a non-negative integer.")

File: php/Petshop.php
<?php
class Petshop
{
    // private attributes for the class
    private string $ID;
    private string $nama;
    private string $kategori;
    private int $harga;
    private string $gambar;

    // constructor for class, also setting each attributes in the process
    function __construct(string $ID, string $nama, string $kategori, int $harga, string $gambar)
    {
        $this->ID = $ID;
        $this->nama = $nama;
        $this->kategori = $kategori;
        $this->harga = $harga;
        $this->gambar = $gambar;
    }

    // Method untuk mengatur atribut ID
    public function set_ID(string $input_ID): void
    {
        $this->ID = $input_ID;
    }

    // Method untuk mengambil atribut ID
    public function get_ID(): string
    {
        return $this->ID;
    }

    // Method untuk mengatur atribut nama
    public function set_Nama(string $input_nama): void
    {
        $this->nama = $input_nama;
    }

    // Method untuk mengambil atribut nama
    public function get_Nama(): string
    {
        return $this->nama;
    }

    // Method untuk mengatur atribut kategori
    public function set_Kategori(string $input_kategori): void
    {
        $this->kategori = $input_kategori;
    }

    // Method untuk mengambil atribut kategori
    public function get_Kategori(): string
    {
        return $this->kategori;
    }

    // Method untuk mengatur atribut harga
    public function set_Harga(int $input_harga): void
    {
        $this->harga = $input_harga;
    }

    // Method untuk mengambil atribut harga
    public function get_Harga(): int
    {
        return $this->harga;
    }

    // Method untuk mengatur atribut gambar
    public function set_Gambar(string $input_gambar): void
    {
        $this->gambar = $input_gambar;
    }

    // Method untuk mengambil atribut gambar
    public function get_Gambar(): string
    {
        return $this->gambar;
    }

    // Menggabungkan masing-masing atribut dan mengembalikan dalam bentuk array
    public function toArray()
    {
        return [
            'id' => $this->ID,
            'nama' => $this->nama,
            'kategori' => $this->kategori,
            'harga' => $this->harga,
            'gambar' => $this->gambar
        ];
    }

    // Destructor
    function __destruct() {}
}
